### DISCUSSION TITLE
Close popover when item within is clicked

### DISCUSSION THREAD

**Discussion by harrygr** [tailwindlabs/headlessui#382] [Help]
Is there away to explicitly close the popover when a child item is clicked, say, if using it as a navigation menu?

In my case I'm using a popover as a nav menu containing a list of React Router `Link`s.

I'm finding that unless a rerender of the parent is triggered then the popover panel remains open and must be dismissed manually.

It might be worth exposing a `close` render prop unless I'm misusing it somehow...

**Comment by amantiwari1** [comment]
any luck?

**Reply by harrygr** [reply]
I ended up just using [menu](https://headlessui.dev/react/menu) instead which auto closes when one of its items is clicked. It seemed more suitable for my use case anyway.

**Reply by amantiwari1** [reply]
tried menu but still no working

**Reply by aburtsev** [reply]
This functionality is available since [1.4.0 version](https://github.com/tailwindlabs/headlessui/blob/main/CHANGELOG.md#headlessuireactv140---2021-07-29)
